### PR TITLE
feat(docs): fleshing out transfer error scenarios

### PR DESCRIPTION
**Pull request by lewisdaly**

### CODE REVIEW

File: docs/transfer/README.md [hunk 1]
```diff
@@ -163,20 +163,86 @@ A PISP can issue a `GET /thirdpartyRequests/{id}/transactions` to find the statu
 
 ## <a name='ErrorConditions'></a>3. Error Conditions
 
+After the PISP initiates the Thirdparty Transaction Request with `POST /thirdpartyRequests/transactions`, the DFSP must send either a `POST /thirdpartyRequests/transcations/{ID}/error` or `PATCH /thirdpartyRequests/transactions/{ID}` callback to inform the PISP of a final status to the Thirdparty Transaction Request.
```
Comment (eoln) [inline docs/transfer/README.md:166]:
```suggestion
After the PISP initiates the Thirdparty Transaction Request with `POST /thirdpartyRequests/transactions`, the DFSP must send either a `PUT /thirdpartyRequests/transcations/{ID}/error` or `PATCH /thirdpartyRequests/transactions/{ID}` callback to inform the PISP of a final status to the Thirdparty Transaction Request.
```

File: docs/transfer/3-3-1-bad-quote-request.puml [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+end note
+
+rnote left of D2 #Light
+**""PUT /thirdpartyRequests/transactions""**
+**""    /00000000-0000-0000-0000-000000000000/error""**
+""FSPIOP-Source: dfspa""
+""FSPIOP-Destination: pispa""
+{
+  "errorInformation": {
+    "errorCode": "6XXX",
```
Comment (eoln) [inline docs/transfer/3-3-1-bad-quote-request.puml:116]: to report PISP errors we reserved 7xxx  code range.  DFSP probably will not want to share detailed quote error with PISP so maybe we should point out here that it's DFSP responsibility to send proper 

already defined error codes: https://github.com/mojaloop/sdk-standard-components/blob/master/src/lib/errors/index.js#L99-L110

PISP errorCode.
```suggestion
    "errorCode": "71xx",
```

File: docs/transfer/README.md (hunk 1 above)
Comment (lewisdaly) [reply docs/transfer/README.md:166]: Oops! Well spotted.

File: docs/transfer/3-3-2-bad-transfer-request.puml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+end note
+
+rnote left of D2 #Light
+**""PUT /thirdpartyRequests/transactions""**
+**""    /00000000-0000-0000-0000-000000000000/error""**
+""FSPIOP-Source: dfspa""
+""FSPIOP-Destination: pispa""
+{
+  "errorInformation": {
+    "errorCode": "6XXX",
```
Comment (eoln) [inline docs/transfer/3-3-2-bad-transfer-request.puml:99]:
```suggestion
    "errorCode": "71xx",
```

File: docs/transfer/3-4-1-bad-signed-challenge-self-hosted.puml [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+
+rnote left of D2 #Light
+**""PUT /thirdpartyRequests/transactions""**
+**""    /00000000-0000-0000-0000-000000000000/error""**
+""FSPIOP-Source: dfspa""
+""FSPIOP-Destination: pispa""
+{
+  "errorInformation": {
+    "errorCode": "6200",
```
Comment (eoln) [inline docs/transfer/3-4-1-bad-signed-challenge-self-hosted.puml:63]:
```suggestion
    "errorCode": "71xx",
```

File: docs/transfer/3-4-2-bad-signed-challenge-auth-service.puml [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+AUTHS -> AUTHS: Check that the signed bytes match the \npublickey we have stored for the consent
+
+rnote right of AUTHS #Light
+**""PUT /thirdpartyRequests/verifications""**
+**""    /44444444-0000-0000-0000-000000000000/error""**
+""FSPIOP-Source: central-auth""
+""FSPIOP-Destination: dfspa""
+{
+  "errorInformation": {
+    "errorCode": "6200",
```
Comment (eoln) [inline docs/transfer/3-4-2-bad-signed-challenge-auth-service.puml:92]:
```suggestion
    "errorCode": "71xx",
```

File: docs/transfer/3-4-2-bad-signed-challenge-auth-service.puml [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+
+rnote left of D2 #Light
+**""PUT /thirdpartyRequests/transactions""**
+**""    /00000000-0000-0000-0000-000000000000/error""**
+""FSPIOP-Source: dfspa""
+""FSPIOP-Destination: pispa""
+{
+  "errorInformation": {
+    "errorCode": "6200",
```
Comment (eoln) [inline docs/transfer/3-4-2-bad-signed-challenge-auth-service.puml:115]:
```suggestion
    "errorCode": "71xx",
```

File: docs/transfer/3-5-user-rejects-tpr.puml [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+
+D2 -> D2: Look up the ""transactionRequestId"" for this ""authorizationId""
+D2 -> D2: See that the user rejected the Thirdparty Transaction Request
+
+rnote over D2 #Light
+**""PATCH /thirdpartyRequests/transactions/00000000-0000-0000-0000-000000000000""**
+""FSPIOP-Source: dfspa""
+""FSPIOP-Destination: pispa""
+{
+  "transactionRequestState": "REJECTED",
```
Comment (eoln) [inline docs/transfer/3-5-user-rejects-tpr.puml:50]: `transactionId` is required property for PATCH

```suggestion
  "transactionId": "1111....",
  "transactionRequestState": "REJECTED",
```

File: docs/transfer/3-2-1-bad-tx-request.puml [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+
+
+rnote left of D2 #Light
+**""PUT /thirdpartyRequests/transactions""**
+**""    /00000000-0000-0000-0000-000000000000/error""**
+""FSPIOP-Source: dfspa""
+""FSPIOP-Destination: pispa""
+{
+  "errorInformation": {
+    "errorCode": "6105",
```
Comment (eoln) [inline docs/transfer/3-2-1-bad-tx-request.puml:64]:
```suggestion
    "errorCode": "7105",
```

**Review by eoln — CHANGES_REQUESTED** [review]
let use already defined error codes: https://github.com/mojaloop/sdk-standard-components/blob/master/src/lib/errors/index.js#L99-L110

File: docs/transfer/3-5-user-rejects-tpr.puml (hunk 7 above)
Comment (lewisdaly) [reply docs/transfer/3-5-user-rejects-tpr.puml:50]: Yeah, but I think we've just found a case where `transactionId` wouldn't exist yet

**Comment by lewisdaly** [comment]
Ah I see @eoln  - I was looking at the wrong error codes doc. I'll update them accordingly.

File: docs/transfer/3-5-user-rejects-tpr.puml (hunk 7 above)
Comment (eoln) [reply docs/transfer/3-5-user-rejects-tpr.puml:50]: nope :-D  `transactionId` already exists. 
It has been allocated before the quote request happens (and quote requires it)
PISP is informed about it by receiving PUT with `transactionRequestState='RECEIVED'` a step before.
User accepts/rejects tpr seeing quote details.

Comment (lewisdaly) [reply docs/transfer/3-5-user-rejects-tpr.puml:50]: Ah - of course! That's a happy accident :)

Comment (eoln) [reply docs/transfer/3-5-user-rejects-tpr.puml:50]: I suggest keeping `transactionId` as a required field for PATCH - the client code will be then simpler -> no need to check for the existence of this property before accessing.

**Comment by lewisdaly** [comment]
@eoln good to go again.

**Review by eoln — APPROVED** [review]

**Pull request merged**